src/libData/AccountData/AccountStoreSC.tpp

```diff
@@ -894,68 +934,110 @@ bool AccountStoreSC<MAP>::ParseCallContractJsonOutput(const Json::Value& _json,
     }
   }
 
-  if (!contractAccount->SetStorage(state_entries, true)) {
-    LOG_GENERAL(WARNING, "SetStorage failed");
-  }
-
-  if (ENABLE_CHECK_PERFORMANCE_LOG) {
-    LOG_GENERAL(DEBUG, "LDB Write (microseconds) = " << r_timer_end(tpStart));
-    LOG_GENERAL(DEBUG, "Gas used = " << (startGas - gasRemained));
-  }
-
   for (const auto& e : _json["events"]) {
     LogEntry entry;
     if (!entry.Install(e, m_curContractAddr)) {
+      receipt.AddError(LOG_ENTRY_INSTALL_FAILED);
       return false;
     }
-    m_curTranReceipt.AddEntry(entry);
+    receipt.AddEntry(entry);
   }
 
+  bool ret = false;
+
   // If output message is null
   if (_json["message"].isNull()) {
     LOG_GENERAL(INFO,
                 "null message in scilla output when invoking a "
                 "contract, transaction finished");
-    return true;
+    ret = true;
   }
 
-  // Non-null messages must have few mandatory fields.
-  if (!_json["message"].isMember("_tag") ||
-      !_json["message"].isMember("_amount") ||
-      !_json["message"].isMember("params") ||
-      !_json["message"].isMember("_recipient")) {
-    LOG_GENERAL(WARNING,
-                "The message in the json output of this contract is corrupted");
-    return false;
-  }
+  Address recipient;
+  Account* account = new Account();
 
-  Address recipient = Address(_json["message"]["_recipient"].asString());
-  if (recipient == Address()) {
-    LOG_GENERAL(WARNING, "The recipient can't be null address");
-    return false;
+  if (!ret) {
+    // Non-null messages must have few mandatory fields.
+    if (!_json["message"].isMember("_tag") ||
+        !_json["message"].isMember("_amount") ||
+        !_json["message"].isMember("params") ||
+        !_json["message"].isMember("_recipient")) {
+      LOG_GENERAL(
+          WARNING,
+          "The message in the json output of this contract is corrupted");
+      receipt.AddError(MESSAGE_CORRUPTED);
+      return false;
+    }
+
+    try {
+      m_curAmount = boost::lexical_cast<boost::multiprecision::uint128_t>(
+          _json["message"]["_amount"].asString());
+    } catch (...) {
+      LOG_GENERAL(WARNING, "_amount " << _json["message"]["_amount"].asString()
+                                      << " is not numeric");
+      return false;
+    }
+
+    recipient = Address(_json["message"]["_recipient"].asString());
+    if (recipient == Address()) {
+      LOG_GENERAL(WARNING, "The recipient can't be null address");
+      receipt.AddError(RECEIPT_IS_NULL);
+      return false;
+    }
+
+    delete account;
+    account = this->GetAccount(recipient);
+
+    if (account == nullptr) {
+      AccountStoreBase<MAP>::AddAccount(recipient, {0, 0});
+    }
+
+    // Recipient is non-contract
+    if (!account->isContract()) {
```

Comment:
Can potentially crash if account == nulltr at line 991

Reply:
GetAccount can give you a nullptr if the account was not created yet, so just Add a new Account obj for it shall be the right approach

Reply:
Is `AccountStoreBase<MAP>::AddAccount(recipient, {0, 0});` going to update local `account*` ?